python/packages/autogen-core/docs/src/user-guide/core-user-guide/cookbook/distributed-group-chat.ipynb

```diff
@@ -0,0 +1,258 @@
+    "    ChatCompletionClient,\n",
+    "    LLMMessage,\n",
+    "    OpenAIChatCompletionClient,\n",
+    "    SystemMessage,\n",
+    "    UserMessage,\n",
+    ")\n",
+    "from pydantic import BaseModel\n",
+    "\n",
+    "\n",
+    "class GroupChatMessage(BaseModel):\n",
+    "    \"\"\"Implements a sample message sent by an LLM agent\"\"\"\n",
+    "\n",
+    "    body: LLMMessage\n",
+    "\n",
+    "\n",
+    "class RequestToSpeak(BaseModel):\n",
+    "    \"\"\"Message type for agents to speak\"\"\"\n",
+    "\n",
+    "    pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Group Chat Manager\n",
+    "- **group chat manager** implements a round-robing algorithm to ask agents to talk, one at a time. It sends `SpeakRequest` requests to agent topics. This agent subscrives to topic type **agent_result** to receive participants messages.\n",
+    "\n",
+    "# Participants\n",
+    "- **data scientist**: Listens to `SpeakRequest` sent by manager and publishes a LLM API call response as `GroupChatMessage`. This agent subscribes to `topic_type=\"data_scientist\", agent_type=\"data_scientist_ag_type\"`\n",
+    "- **art director**:  Listens to `SpeakRequest` sent by manager and publishes a LLM API call response as `GroupChatMessage`. This agent subscribes to `topic_type=\"art_director\", agent_type=\"art_director_ag_type\"`\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ParticipantAgent(RoutedAgent):\n",
+    "    def __init__(self, name: str, description: str, model_client: ChatCompletionClient, system_message: str) -> None:\n",
+    "        super().__init__(\"A Participant\")\n",
+    "        self._name = name\n",
+    "        self._description = description\n",
+    "        self._model_client = model_client\n",
+    "        self._chat_history: List[LLMMessage] = [SystemMessage(system_message)]\n",
+    "\n",
+    "    @message_handler\n",
+    "    async def handle_message(self, message: GroupChatMessage, ctx: MessageContext) -> None:\n",
+    "        print(f\"received {message}\")\n",
+    "        self._chat_history.append(message.body)\n",
+    "\n",
+    "    @message_handler\n",
+    "    async def handle_request_to_speak(self, message: RequestToSpeak, ctx: MessageContext) -> None:\n",
+    "        completion = await self._model_client.create(self._chat_history)\n",
+    "        assert isinstance(completion.content, str)\n",
+    "        self._chat_history.append(AssistantMessage(content=completion.content, source=self._name))\n",
+    "        print(f\"\\n{'-'*80}\\n{self._name} ({id(self)}):\\n{completion.content}\")\n",
+    "        await asyncio.sleep(2)\n",
+    "        await self.publish_message(\n",
+    "            GroupChatMessage(body=UserMessage(content=completion.content, source=self._name)),\n",
+    "            TopicId(type=\"agent_result\", source=\"participant\"),\n",
```

Comment:
Instead of specifying the `source`, either use the agent's ID key or use `DefaultTopicId(topic_type=...)`

Reply:
This is for multi-tenancy consideration.